cmd/buildah/rmi.go

```diff
@@ -47,26 +57,58 @@ func rmiCmd(c *cli.Context) error {
 		return err
 	}
 
-	for _, id := range args {
-		image, err := getImage(id, store)
+	imagesToDelete := args[:]
+	var lastError error
+
+	if removeAll {
+		images, err := store.Images()
 		if err != nil {
-			return errors.Wrapf(err, "could not get image %q", id)
+			return errors.Wrapf(err, "error reading images")
+		}
+		for _, image := range images {
+			imagesToDelete = append(imagesToDelete, image.ID)
+		}
+	}
+
+	for _, id := range imagesToDelete {
+		image, err := getImage(id, store)
+		if err != nil || image == nil {
+			if lastError != nil {
+				fmt.Fprintln(os.Stderr, lastError)
+			}
+			if err == nil {
+				err = errors.New("Image does not Exist")
```

Comment:
How about.
err=storage.ErrNotAnImage

Reply:
Sold, I'll also change the errors.New a few lines down to use storage.ErrImageUsedByContainer instead of Container Exists.